fda-model/structures/Inventory.cs

```diff
@@ -0,0 +1,156 @@
+        {
+            get
+            {
+                List<string> impactAreas = new List<string>();
+                foreach (var structure in Structures)
+                {
+                    if (!impactAreas.Contains(structure.ImpactAreaID.ToString()))
+                    {
+                        impactAreas.Add(structure.ImpactAreaID.ToString());
+                    }
+                }
+                return impactAreas;
+            }
+        }
+
+        /// <summary>
+        /// Loops through entire inventory and reports back a list of all the unique damage catagories associated with the structures
+        /// </summary>
+        /// <returns></returns>
+        public List<string> GetUniqueDamageCatagories()
+        {
+            List<string> damageCatagories = new List<string>();
+            foreach (Structure structure in Structures)
+            {
+                if (damageCatagories.Contains(structure.DamCatName))
+                {
+                    continue;
+                }
+                else
+                {
+                    damageCatagories.Add(structure.DamCatName);
+                }
+            }
+            return damageCatagories;
+        }
+
+
+        /// <summary>
+        /// Constructor to create a SI from a shapefile. Gonna need to do this from database potentially as well
+        /// </summary>
+        /// <param name="pointShapefilePath"></param>
+        public Inventory(string pointShapefilePath, string impactAreaShapefilePath)
+        {
+            PointFeatureLayer structureInventory = new PointFeatureLayer("Structure_Inventory", pointShapefilePath);
+            PointMs pointMs = new PointMs(structureInventory.Points().Select(p => p.PointM()));
+            _structures = new List<Structure>();
+            for (int i = 0; i < structureInventory.FeatureCount(); i++)
+            {
+
+                PointM point = pointMs[i];
+                var row = structureInventory.FeatureRow(i);
+                int fid = (int)row["fd_id"];
+                double found_ht = (double)row["found_ht"];
+                double val_struct = (double)row["val_struct"];
+                double val_cont = (double)row["val_cont"];
+                double val_vehic = (double)row["val_vehic"];
+                string st_damcat = (string)row["st_damcat"];
+                string occtype = (string)row["occtype"];
+                int pop2amu65 = (int)row["pop2amu65"];
+                int pop2amo65 = (int)row["pop2amo65"];
+                int pop2pmu65 = (int)row["pop2pmu65"];
+                int pop2pmo65 = (int)row["pop2pmo65"];
+                string cbfips = (string)row["cbfips"];
+                int impactAreaID = GetImpactAreaID(point, impactAreaShapefilePath);
+                _structures.Add(new Structure(fid, point, found_ht, val_struct, val_cont, val_vehic, st_damcat, occtype, pop2amu65, pop2amo65, pop2pmu65, pop2pmo65, impactAreaID, cbfips));
+            }
```

Comment:
we'll also need to track ground stage, beginning damage depth, number of structures, year in service, notes/meta data (may not be needed in the model)